### DISCUSSION TITLE
Use icons from theme

### DISCUSSION THREAD

**Discussion by kybos** [muse-sequencer/muse#1142] [Team Posts]
This option causes inconsistent UI look in MusE on many recent Linux systems, because most default themes currently prefer the single-color icons that don't fit in MusE at all. There are only a few icons affected, but they are just the most visible ones (new document, file open/save etc.).
I personally would prefer to remove the option completely (with another positive side-effect: it would allow for further simplifying and cleaning up the icon handling). 
However, if you think that it should be kept (for backwards compatibility or some other reason), could we at least make "false" the default?

**Comment by terminator356** [comment]
No problem. Go ahead and set false.
I'm to blame for turning the default to true. After I restored a few usages of it, I thought
 might as well, just "because I could".
 
As for removing it, Orcan added it almost ten years ago.
Back when things were a bit more... predictable, when things weren't so... bizarre sometimes.
I trust he would not mind.
I'd say yes. Any objections ? Robert ? Orcan ?
Sad to see them go but...

PS: You any good at stylesheet design? Fancy a wee side fix?
The new rack scrollbar, I just grabbed from the help scrollbar examples and compacted it, verbatim.
But I didn't have time to study what all the strange options are for drawing scrollbars.
Arrows would be nice, they mention you can do that.
I installed new up/down arrow SVGs yesterday.

**Comment by terminator356** [comment]
BTW The stylesheet code is embedded in Rack ctor.
I think we will need to break that out and put it into a global place or a file,
 so that other things can use it. And they will. The other rack spaces will need it too.
A nice skinny scrollbar will become useful.
I'm wondering now whether the sheet can accommodate both horizontal and vertical scroll bars.
The examples seem to indicate that it's one or the other.
So we may need two sheets?

**Comment by kybos** [comment]
Thanks. I will wait in case somebody objects. 
As for the stylesheets: I will give it a try (although the first thing I did was to set the default to 8 so they are displayed completely - looks perfectly OK).

**Comment by kybos** [comment]
BTW: I already asked in the forum if the effect rack colors could be customized by stylesheets as well, I tried but failed, probably not possible because of the special painting routine? Would be nice to be able to do it.

**Comment by terminator356** [comment]
Got the message. Will try...

**Comment by kybos** [comment]
I guess you did not want to use the standard scrollbar because it's too wide?
I played a little with the css, what about something like this:
![image](https://user-images.githubusercontent.com/10009541/70861365-33b96980-1f2d-11ea-8804-b9d716db3436.png)
Best to put it into the external stylesheet (also in the light of Qt6 bound to deprecate the setStylesheeet method)? Then it could be adjusted for each color scheme separately if necessary. But the default scheme currently does not use qss, correct?

**Comment by terminator356** [comment]
Very nice indeed! Awesome.
There were two problems: The width, and the fact that some styles want to put
 the **three** arrows there. Not nearly enough vertical space for that.
I was amazed that with the small version, even with the setting on just **one** visible item,
 the two tiny up/down areas still work. Hopefully if you kept the dimensions of those
 two areas the same, it will still work really cool with just one item visible.

**Comment by kybos** [comment]
Now it should also work with one item only:
![image](https://user-images.githubusercontent.com/10009541/70888108-d3373480-1fdf-11ea-9876-fa324ff128a4.png)
I will put the qss into a ressource file for the time being so it's not in the code, but IMO it would be preferable to put it into the external stylesheets for each color scheme (would probably require creating one for the default scheme as well). I will create a PR, you can check there.

I noticed a weird thing in this context: 
In Appearance, removing the custom stylesheet from the textedit control (with the Undo button) does not remove it, instead it inserts another stylesheet from a resource file (:/style.qss). Additionally, both this stylesheet and the resource file seem to be obsolete?

**Comment by kybos** [comment]
Can I remove this obsolete stylesheet and resource (see above)? Looks like a clear mistake that it is inserted at removing the user stylesheet (and comments in the :/style.qss itself suggest that it was meant to be deleted anyway).

**Comment by terminator356** [comment]
Yes.

History: When I added some stylesheet support way back, I simply grabbed that qss file
 from Werner's original MusE-2 (now called evolution in our attic).
That is where that mysterious file comes from.
It is left over from my first attempts at this stuff.

A few other things come from that version: The AL and AWL libraries (and namespaces),
 because the Deicsonze synth used them, and that synth was just too cool to let rot in
 unmaintained MusE-2 evolution. So I imported Deicsonze. We had an older weaker version.
